tests/sentry/incidents/test_logic.py

```diff
@@ -1137,3 +1144,143 @@ def test_no_excluded(self):
+            trigger,
+            label=label,
+            threshold_type=threshold_type,
+            alert_threshold=alert_threshold,
+            resolve_threshold=resolve_threshold,
+        )
+        assert trigger.label == label
+        assert trigger.threshold_type == threshold_type.value
+        assert trigger.alert_threshold == alert_threshold
+        assert trigger.resolve_threshold == resolve_threshold
+
+    def test_name_used(self):
+        label = "uh oh"
+        create_alert_rule_trigger(self.alert_rule, label, AlertRuleThresholdType.ABOVE, 1000, 400)
+        trigger = create_alert_rule_trigger(
+            self.alert_rule, "something else", AlertRuleThresholdType.ABOVE, 1000, 400
+        )
+        with self.assertRaises(AlertRuleTriggerLabelAlreadyUsedError):
+            update_alert_rule_trigger(trigger, label=label)
+
+    def test_exclude_projects(self):
+        other_project = self.create_project()
+
+        alert_rule = self.create_alert_rule(projects=[other_project, self.project])
+        trigger = create_alert_rule_trigger(
+            alert_rule, "hi", AlertRuleThresholdType.ABOVE, 1000, 400
+        )
+        update_alert_rule_trigger(trigger, excluded_projects=[other_project])
+        assert trigger.exclusions.get().query_subscription.project == other_project
+
+    def test_complex_exclude_projects(self):
+        excluded_project = self.create_project()
+        other_project = self.create_project()
+
+        alert_rule = self.create_alert_rule(
+            projects=[excluded_project, self.project, other_project]
+        )
+        trigger = create_alert_rule_trigger(
+            alert_rule,
+            "hi",
+            AlertRuleThresholdType.ABOVE,
+            1000,
+            400,
+            excluded_projects=[excluded_project, self.project],
+        )
+        update_alert_rule_trigger(trigger, excluded_projects=[other_project, excluded_project])
+        excluded_projects = [
+            exclusion.query_subscription.project for exclusion in trigger.exclusions.all()
+        ]
+        assert set(excluded_projects) == set([other_project, excluded_project])
+
+    def test_excluded_projects_not_associated_with_rule(self):
+        other_project = self.create_project()
+        alert_rule = self.create_alert_rule(projects=[self.project])
+        trigger = create_alert_rule_trigger(
+            alert_rule, "hi", AlertRuleThresholdType.ABOVE, 1000, 400
+        )
+
+        with self.assertRaises(ProjectsNotAssociatedWithAlertRuleError):
+            update_alert_rule_trigger(trigger, excluded_projects=[other_project])
+
+
+class DeleteAlertRuleTriggerTest(TestCase):
+    def test(self):
+        trigger = create_alert_rule_trigger(
+            self.create_alert_rule(), "hi", AlertRuleThresholdType.ABOVE, 1000, 400
+        )
+        trigger_id = trigger.id
+        delete_alert_rule_trigger(trigger)
+
+        assert not AlertRuleTrigger.objects.filter(id=trigger_id).exists()
```

Comment:
Is there any value in asserting that alert rule trigger exclusions are also removed?

Reply:
Hmm, I'm not really sure. These cascades are just part of Django, not sure if there's value in testing that it's working? On the other hand it is behaviour we're relying on.